### ISSUE TITLE
Doesn't work for Java 8

### ISSUE THREAD

**Issue by nedtwigg** [youribonnaffe/gradle-format-plugin#8]
Great plugin!  Thank you!

It doesn't work for formatting Java 8 at the moment, and it fails silently.  Looking at [JavaFormatter.java](https://github.com/hibernate/hibernate-tools/blob/master/src/java/org/hibernate/tool/ide/formatting/JavaFormatter.java#L66-68), it looks like you probably want to [check the return value](https://github.com/youribonnaffe/gradle-format-plugin/blob/master/src/main/groovy/com/github/youribonnaffe/gradle/format/FormatTask.groovy#L22).

There's a super simple test case available here: https://github.com/diffplug/durian

The example code is already formatted correctly, so if you change the spacing, then run `gradlew format`, it should change back.  You'll see that it doesn't.  However, if you go into `format_java.properties` and find-replace `1.8` with `1.7`, then it does change back.  Looks like you're using an older version of the Eclipse formatter.

There have been a lot of improvements which are available in the latest 4.5 M6:
[https://bugs.eclipse.org/bugs/show_bug.cgi?id=303519](Eclipse bug tracker)
[https://waynebeaton.wordpress.com/2015/03/15/great-fixes-for-mars-winners-part-i/](Broad description)

It looks like hibernate tools isn't providing much value for your use case, and you could update to a newer formatter and reduce the download size of your plugin if you made your own `JavaFormatter` instance.  In that case your dependency would be [this](https://bintray.com/bintray/jcenter/org.eclipse.tycho%3Aorg.eclipse.jdt.core/3.10.0.v20140604-1726/view) instead of Hibernate tools.

**Comment by youribonnaffe** [comment]
Thanks for spotting that, I'll try to take a look. Contributions are welcome too :)
I must confess I took the same dependency as we used before in a Ant task (it seemed quite hard to find the latest Eclipse dependency on Maven central).

**Comment by nedtwigg** [comment]
Took me a while to find the Eclipse depenency myself!  I found it by digging through the pom for hibernate tools.

Internally, we have a plugin that downloads a whole Eclipse install, and runs the source code formatter using some command line arguments to eclipse.exe.  It works for us because we also use that downloaded Eclipse for a bunch of other things.  We're releasing some of our code as opensource, and we don't want to burden our tiny library with this huge install.

TLDR; I won't have time to contribute this feature any time soon, and don't expect you to either!  But eventually one of us might :)

**Comment by youribonnaffe** [comment]
@nedtwigg you create issue #8 for a Java 8 problem :) however I could not reproduce it :(.
I added a test case trying to simulate what you observed : 5f0a38a1e266b2fb479cb56bd998659ece25d81d

I took your suggestion to remove Hibernate tools and depend directly on Eclipse.

**Comment by nedtwigg** [comment]
That's because you already fixed it by going to `org.eclipse.tycho:org.eclipse.jdt.core:3.10.0.v20140604-1726`.  Hibernate tools used a much older formatter, which didn't support Java 8 and was also much slower.

Huge thanks, can't wait for the release!  Now you're only a few lines away from implementing checking as well :D !

**Comment by youribonnaffe** [comment]
I actually did run the test before upgrading the dependency and it seemed to work

**Comment by nedtwigg** [comment]
Hmmm...  I just confirmed my original bug report, and it still fails as descibed, and changing to `1.7` still fixes it.  I checked out your branch and cherry-picked the test, and confirmed that your test does pass with `1.8` even before the upgrade.

I thought maybe it only fails if the code in question actually uses Java 1.8, but that's not right because my test case didn't actually use any Java 8 features...

Then I tried changing your test code to something just a teensy bit more complicated:
```java
public enum SomeEnum {
A,
 B,
  C;
}
```
and the formatter refuses format it with 1.7 properties or 1.8!  It always returns a null `TextEdit`, indicating that it's somehow failing to parse this simple enum.

I'm not sure what's going on, but maybe the formatter requires different arguments than the old one does, but it can get by a simple `static void main()` class.  It's also possible that there's some Eclipse jar that's missing from the classpath, or some OSGI service that needs to get manually configured in this newer version.

I'm somewhat familiar with Eclipse internals, I'll do some stepping through the formatter tonight to figure out what's going on.  In the meantime, I wouldn't ship this yet because something seems to be broken.

**Comment by nedtwigg** [comment]
Just wanted to let you know the final conclusion I took.  I forked to a new project called [spotless](https://github.com/diffplug/spotless).  It's based around the idea that a formatter should take a String as input, and return another String as output.  By making this core model simple, it was easy to implement things like the the `check` task, lazy loading, and composable check tasks.

Feel free to pull anything you'd like from it, and lemme know if there are any changes in attribution you'd like me to make in spotless.

**Comment by youribonnaffe** [comment]
Thanks @nedtwigg , I took a few things from your code, might take inspiration from the check task too.
Also see #10 for the enum/Java 8 formatting issue.

**Comment by nedtwigg** [comment]
Great!  Good detective work on the core-3.1.1 issue.
